doc/iterations/2024.1/shapes/add_support_for_libraries.adoc

```diff
@@ -0,0 +1,39 @@
+= Add the support for libraries
+
+== Context
+
+Users want to have access to read only data (a.k.a libraries).
+
+== Key Result
+
+In the Details View, users could reference these read only data from their models' elements.
+
+== Solution
+
+1. When declaring templates, specifiers should be able to specify a list of libraries that will be accessible in the projects created from the templates.
+
+2. Specifiers should be able to associate libraries to projects' natures. Thus, when a project with a specific nature is created/uploaded, the libraries associated to this specific nature will be accessible.
+
+3. The `Blank project` has no nature and is not a template. So with the proposed solution, it will not be possible to use libraries in a `Blank project`. To handle this specific case of the `Blank project`, it should be possible to add libraries to every project without nature. This will be done programmatically, by the specifier.
+
+These libraries are models stored as files.
+Then, in the Details View, users could reference these read only data from their models' elements.
+Furthermore, specifiers will be able to reference these read only data programmatically (e.g. when invoking creation tools, set operations,...).
+
+=== Breadboarding
+
+Same UI as before.
+
+=== Cutting backs
+
+== Rabbit holes
+
+A template may add a lot of libraries, and thus a lot a model elements.
+These model elements will be accessible in the reference widgets from the Detail view.
+In some case the list of elements accessible in the reference widgets will be huge.
+Specifiers may have to programmatically filter the candidates of reference widgets.
+Note that this is also the case without the introduction of libraries, but libraries may emphasize the case.
```

Comment:
I'm not sure that's related at all with this issue. We could talk more to say on "how to keep people from modifying those read only objects" for example.

Reply:
That's why I've added the last sentence. The problem raised is not introduced by the libraries may emphasize it.